### PR TITLE
[qtpositioning] devendor poly2tri

### PR DESCRIPTION
**Pull request by m-kuhn**
Devendor poly2tri from `qtpositioning`.

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

**Comment by dg0yt** [comment]
The patch is fairly large simply due to removal of files, and this part will be sensitive to souce changes. I would suggest to trim the patch and to do file removal as needed with cmake commands.

**Comment by m-kuhn** [comment]
> The patch is fairly large simply due to removal of files, and this part will be sensitive to souce changes. I would suggest to trim the patch and to do file removal as needed with cmake commands.

Good point. How could this be done?
The download happens in `qt_install_submodule` in the `qtbase` port.

- Patch `qt_install_submodule` (already done for `qttools` / `qtwebengine`)
- Reimplement `qt_install_submodule` in `qtpositioning`
- Implement some sort of callback mechanism or a `SRC_DIRECTORIES_TO_REMOVE` param

File: ports/qtpositioning/devendor-poly2tri.patch [hunk 1]
```diff
@@ -0,0 +1,2938 @@
...
+-
+-#endif
+diff --color -Naur a/src/CMakeLists.txt b/src/CMakeLists.txt
+--- a/src/CMakeLists.txt	2023-05-13 16:59:42.379052157 +0200
++++ b/src/CMakeLists.txt	2023-05-13 17:00:19.256085781 +0200
+@@ -2,7 +2,6 @@
+ # SPDX-License-Identifier: BSD-3-Clause
+ 
+ # special case begin
+-add_subdirectory(3rdparty/poly2tri)
```
Comment (Neumann-A) [inline ports/qtpositioning/devendor-poly2tri.patch:2935]: Is there a reason this line is not enough ?

Comment (m-kuhn) [reply ports/qtpositioning/devendor-poly2tri.patch:2935]: Only for me to be sure during development that there's no reference left somewhere. But effectively, that's the most simple version.

File: ports/qtpositioning/devendor-poly2tri.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ add_subdirectory(3rdparty/clipper)
+ add_subdirectory(3rdparty/clip2tri)
```
[the remark this answers is not in the document]
Comment (LilyWangLL) [reply ports/qtpositioning/devendor-poly2tri.patch:36]: Ping @m-kuhn for response.

[the remark this answers is not in the document]
Comment (m-kuhn) [reply ports/qtpositioning/devendor-poly2tri.patch:36]: We could, but I don't have an immediate plan for that

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**